### PR TITLE
fix: align volcano threshold behavior

### PR DESCRIPTION
**Pull request by TJoshMeyer**
## Changes

This PR resolves threshold-consistency issues identified from Issue #249 across MOSuite, including Volcano and Differential filtering paths.

The changes:

- Align Enhanced Volcano threshold semantics with strict comparisons:
  - p-value is significant when p < signif_threshold
  - absolute fold change is significant when abs(logFC) > change_threshold
- Keep Enhanced volcano labeling consistent with strict classification at boundary values.
- Replace positional threshold references with resolved column-name access in Enhanced Volcano classification logic.
- Pass explicit per-feature colors to EnhancedVolcano using resolved logFC and significance columns.
- Apply the same strict threshold semantics when Summary constructs significant-feature output.
- Extend threshold consistency to Differential filtering behavior so fold-change and significance comparisons follow the same strict policy package-wide.
- Update user-facing threshold wording/messages where needed to reflect strict comparisons.
- Document threshold behavior in Volcano roxygen documentation.
- Add boundary-value regression tests for features exactly on p-value and fold-change cutoffs.
- Keep Enhanced responsible for per-contrast visualization and Summary responsible for combined multi-contrast composition.

## Issues

Towards #249 (does not fully address the issue)

## PR Checklist

(~Strikethrough~ any points that are not applicable.)

- [x] This comment contains a description of changes with justifications, with any relevant issues linked.
- [x] Write unit tests for any new features, bug fixes, or other code changes.
- [x] Update the docs if there are any API changes (roxygen2 comments, vignettes, readme, etc.).
- [x] Update `NEWS.md` with a short description of any user-facing changes and reference the PR number. Follow the style described in <https://style.tidyverse.org/news.html>
- [x] Run `devtools::check()` locally and fix all notes, warnings, and errors.

### CODE REVIEW

**Comment by TJoshMeyer** [comment]
This update extends the threshold-consistency fix beyond volcano-only paths to ensure consistency across MOSuite where the same threshold semantics apply.

What was added in this update:

- Differential filtering logic now uses the same strict threshold policy already adopted for volcano behavior:
- significance uses p < cutoff
- fold change uses abs(logFC) > cutoff
- Enhanced volcano label behavior remains aligned with strict boundary handling.
- Boundary-focused regression coverage remains in place for threshold edge cases.

Validation notes:

- Focused volcano + differential tests pass in the unsandboxed local environment.
- Earlier failures in sandboxed runs were due to environment/path restrictions, not threshold-logic regressions.

Issue linkage:

- This remains part of the same fix scope and still addresses Issue 249 under PR 266.

**Comment by TJoshMeyer** [comment]
### Threshold comparator choice rationale

I want to document the rationale for the threshold comparator policy used in this PR.

After reviewing relevant literature and polling colleagues, I did not find a hard-and-fast rule requiring inclusive (<=, >=) versus exclusive (<, >) threshold comparisons for this workflow. Based on that, I chose the more conservative exclusive policy for consistency across MOSuite thresholding paths:

- significance: p < cutoff
- fold change: abs(logFC) > cutoff

This was a deliberate workflow-alignment choice in the absence of compelling evidence that one comparator policy is universally required, and _should not be interpreted as a claim of definitive best practice_.

**Comment by TJoshMeyer** [comment]
### Local verification status update before requesting review:

I ran devtools::check() locally in a validated environment (including Quarto setup). Current status is:

- 0 errors
- 0 warnings
- 4 notes

The remaining notes are:

1. hidden files/directories under test-generated figure output paths
2. non-portable long temp-derived file paths from local test artifacts
3. non-standard top-level local working files in my checkout
4. non-standard files in the temporary R CMD check directory (report.qmd, results)

These appear to be local environment/artifact notes rather than regressions in package logic, so I’m proceeding with code review.

Could maintainers advise whether this note profile (i.e. calling for code review in spite of these kinds of Notes) is acceptable in future PRs, or whether there is a preferred cleanup/remediation pattern we should adopt to avoid them consistently?

**Review by kelly-sovacool — CHANGES_REQUESTED** [review]

**Comment by kelly-sovacool** [comment]
Thanks Josh, will re-review shortly.

P.S. you can leave out the below type of content from PR comments, I can see those details (latest commit hash, github check status) right here in the conversation history

> The current PR head is [351736e](https://github.com/CCBR/MOSuite/commit/351736e25a1b9f86853e0f465e20aa49cc56d521), and all GitHub checks are passing.

File: tests/testthat/test-plot_volcano_enhanced.R [hunk 1]
```diff
@@ -84,6 +84,96 @@ test_that("plot_volcano_enhanced forwards shared styling parameters", {
...
+    tracer = quote(options(
+      mosuite_test_volcano_boundary_args = append(
+        getOption("mosuite_test_volcano_boundary_args"),
+        list(list(colCustom = colCustom))
+      )
+    )),
+    print = FALSE
+  )
+  on.exit(untrace(EnhancedVolcano::EnhancedVolcano), add = TRUE)
+  on.exit(options(mosuite_test_volcano_boundary_args = NULL), add = TRUE)
```
[the remark this answers is not in the document]
Comment (kelly-sovacool) [reply tests/testthat/test-plot_volcano_enhanced.R]: This type of complicated code is a sign that the functionality that this is trying to test should be its own helper function. I created a PR to refactor this for simplicity, see #271

**Review by kelly-sovacool — CHANGES_REQUESTED** [review]
Take a look at #271 and approve it if you agree that it's a good change. Once that's merged into this branch, this one will be ready to go.

**Review by kelly-sovacool — APPROVED** [review]
